### PR TITLE
AssignmentController tests

### PR DESCRIPTION
**Pull request by k3ntako**

### CODE REVIEW

File: server/controllers/AssignmentController/create.js [hunk 1]
```diff
@@ -24,29 +24,15 @@ module.exports = async (req, res, next) => {
...
-
-    if(name.length > 100){
-      throw new ValidationError('Assignment name cannot be longer than 100 characters');
-    }
-
-    let assignee_limit = req.body.assigneeLimit;
-    if (!Number.isInteger(assignee_limit) || assignee_limit < 1){
-      throw ValidationError('Assignee limit must be an integer greater than 0');
-    }
-
```
Comment (k3ntako) [inline server/controllers/AssignmentController/create.js]: assignment validation will happen in the model - https://github.com/mighty-speed-krush/Mighty-Speed-Krush/pull/176

File: server/controllers/AssignmentController/update.js [hunk 2]
```diff
@@ -28,17 +25,24 @@ module.exports = async (req, res, next) => {
       throw new NotFoundError('Assignment with specified ID does not exist on this season');
     }
```
Comment (mienaikoe) [inline server/controllers/AssignmentController/update.js]: You don't necessarily need to do this. You can supply join logic in the update call, either with include or manual sql strings.

File: server/tests/AssignmentController/assign.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+const categoryName = "AssignmentController Assign Category";
+const assignmentName = "AssignmentController Assign Assignment";
+
+module.exports = () => {
+  before(async () => {
+    coachSession = TestGlobal.getGlobalSession();
+    memberSession1 = await TestHelper.join("AssignmentController_assign@wakebird.com");
+    memberSession2 = await TestHelper.join("AssignmentController_assign@wakebird.com");
+    memberSession3 = await TestHelper.join("AssignmentController_assign@wakebird.com");
+
```
Comment (mienaikoe) [inline server/tests/AssignmentController/assign.js]: These all have the same email. Not sure how that works.

File: server/controllers/AssignmentController/_assigner.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ const deleteReplacePerson = async (transaction, assignment_id, replace_person_id
   }
 }
 
-module.exports = async (seasonId, assignment_id, person_id, replace_person_id) => {
+module.exports = async (seasonId, assignment_id, person_id, replace_person_id = null) => {
   await sequelize.transaction(async transaction => {
     if (!person_id) {
       throw new ValidationError("No person ID provided");
```
Comment (mienaikoe) [inline server/controllers/AssignmentController/_assigner.js]: can't remember if you had already resovled this, but validation should be outside the transaction

Comment (k3ntako) [reply server/controllers/AssignmentController/_assigner.js]: it was fixed in a different branch, so I had to merge that in

File: server/tests/AssignmentController/assign.js (hunk 3 above)
Comment (k3ntako) [reply server/tests/AssignmentController/assign.js]: yeah definitely doesn't work. maybe we need a check for this in the test helper.

File: server/controllers/AssignmentController/update.js (hunk 2 above)
Comment (k3ntako) [reply server/controllers/AssignmentController/update.js]: Use Sequlize.literal instead when you update. Check that the season id = req.session.season_id

File: server/controllers/AssignmentController/update.js [hunk 5]
```diff
@@ -7,39 +7,32 @@ module.exports = async (req, res, next) => {
...
       name,
       description,
       assignee_limit,
       assignment_category_id,
       person_id,
     }, {
       where: {
-        id: assignment.id,
+        id: req.params.assignment_id,
+        assignment_category_id: {
```
Comment (mienaikoe) [inline server/controllers/AssignmentController/update.js]: `person_id: req.session.person_id`

File: server/controllers/AssignmentController/update.js [hunk 6]
```diff
@@ -7,39 +7,32 @@ module.exports = async (req, res, next) => {
...
-          },
-        }],
-      }],
-    });
-
-    if (!assignment) {
-      throw new NotFoundError('Assignment with specified ID does not exist on this season');
-    }
-
     const { name, description, assignee_limit, assignment_category_id, person_id } = req.body;
```
Comment (mienaikoe) [inline server/controllers/AssignmentController/update.js]: if req.session.is_coach
then don't worry about filtering by person_id

File: server/controllers/AssignmentController/update.js (hunk 5 above)
Comment (mienaikoe) [reply server/controllers/AssignmentController/update.js]: Coach or Owner of assignment

**Pull request merged**